jenkins/databricks/build.sh

```diff
@@ -15,437 +15,303 @@
+    artifacts[JACKSONCORE]="-DgroupId=com.fasterxml.jackson.core -DartifactId=jackson-core"
+    dep_jars[JACKSONCORE]=${PREFIX_WS_SP_MVN_HADOOP}--com.fasterxml.jackson.core--jackson-databind--com.fasterxml.jackson.core__jackson-databind__${sw_versions[FASTERXML_JACKSON]}.jar
+    artifacts[JACKSONANNOTATION]="-DgroupId=com.fasterxml.jackson.core -DartifactId=jackson-annotations"
+    dep_jars[JACKSONANNOTATION]=${PREFIX_WS_SP_MVN_HADOOP}--com.fasterxml.jackson.core--jackson-annotations--com.fasterxml.jackson.core__jackson-annotations__${sw_versions[FASTERXML_JACKSON]}.jar
+    artifacts[AVROSPARK]="-DgroupId=org.apache.spark -DartifactId=spark-avro_${SCALA_VERSION}"
+    dep_jars[AVROSPARK]=${PREFIX_SPARK}--vendor--avro--avro-${HIVE_VER_STRING}__hadoop-${sw_versions[HADOOP]}_${SCALA_VERSION}_deploy_shaded.jar
+    artifacts[AVROMAPRED]="-DgroupId=org.apache.avro -DartifactId=avro-mapred"
+    dep_jars[AVROMAPRED]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.avro--avro-mapred--org.apache.avro__avro-mapred__${sw_versions[AVRO]}.jar
+    artifacts[AVRO]="-DgroupId=org.apache.avro -DartifactId=avro"
+    dep_jars[AVRO]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.avro--avro--org.apache.avro__avro__${sw_versions[AVRO]}.jar
+
+    # spark-3.1.2 overrides some jar naming conventions
+    if [[ $BASE_SPARK_VERSION == "3.1.2" ]]
+    then
+        dep_jars[HIVE]=${PREFIX_SPARK}--sql--hive--hive_${SCALA_VERSION}_deploy_shaded.jar
+        dep_jars[PARQUETFORMAT]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.parquet--parquet-format--org.apache.parquet__parquet-format__2.4.0.jar
+        dep_jars[AVROSPARK]=${PREFIX_SPARK}--vendor--avro--avro_${SCALA_VERSION}_deploy_shaded.jar
+        dep_jars[AVROMAPRED]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.avro--avro-mapred-hadoop2--org.apache.avro__avro-mapred-hadoop2__${sw_versions[AVRO]}.jar
+        dep_jars[AVRO]=${PREFIX_WS_SP_MVN_HADOOP}--org.apache.avro--avro--org.apache.avro__avro__${sw_versions[AVRO]}.jar
+    fi
+}
+
+# Install dependency jars to MVN repository and build the RAPIDS plugin using mvn package command.
+run_mvn_cmd()
+{
+    # Please note we are installing all of these dependencies using the Spark version
+    # (SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS) to make it easier to specify the dependencies in
+    # the pom files
+    for key in "${!artifacts[@]}"; do
+        echo "running mvn command for $key..."
+        $MVN_CMD -B install:install-file \
+            -Dmaven.repo.local=$M2DIR \
+            -Dfile=$JARDIR/${dep_jars[$key]} \
+            ${artifacts[$key]} \
+            -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
+            -Dpackaging=jar
+    done
+
+    # Build the RAPIDS plugin by running package command for databricks
+    mvn -B -Ddatabricks -Dbuildver=$BUILDVER clean package -DskipTests $MVN_OPT
+}
+
+##########################
+# Main Script starts here
+##########################
+
+initialize
```

Comment:
Since we are reorganizing this I have been wanting to make this so we could skip the maven install commands on rerun. That way if dev is iterating then can just skip it and do only what is required.

I would prefer to see the mvn install stuff in a different function then the actual build command. It also feels weird to me to have the set_ functions calls out in main if nothing other then the mvn install needs them. Perhaps make a dep_install function and call the set_* from there along with the mvn install itself.  Unless I'm missing something else that needs it here.